recipes-ni/nisdbootconfig/files/nisdbootconfig

```diff
@@ -8,13 +8,9 @@ if ! cat /proc/mtd | grep -qs mtd ; then
     sed -i '/ubifs/d' /etc/fstab
     echo LABEL=nibootfs         /boot                ext4       sync           0  0 >>/etc/fstab
     echo LABEL=niconfig         /etc/natinst/share   ext4       sync           0  0 >>/etc/fstab
+    echo LABEL=nirootfs         /mnt/userfs          ext4       defaults       0  0 >>/etc/fstab
     mkdir -p /etc/natinst/share
-
-    # configure the nirootfs partition if we are in safemode
-    if [ -f /etc/natinst/safemode ]; then
-        echo LABEL=nirootfs         /mnt/userfs          ext4       defaults       0  0 >>/etc/fstab
-        mkdir -p /mnt/userfs
-    fi
+    mkdir -p /mnt/userfs
```

Comment:
Why is it appropriate to remove this safemode check? Does the SystemImage postinst script get executed from the safemode context, rather than from a chroot into runmode?

Reply:
On previous implementation, we run nisdbootconfig for both safemode and runmode. Since we do not want to mount /mnt/userfs in runmode, I added the safemode check to isolate this portion out when in runmode. Now that we only run nisdbootconfig in safemode, I thought the safemode check is redundant here.

I believe there's a mix of both in the postinst script. There are some lines where dir manipulation (mainly in /mnt/userfs) is done which I think is in the context of safemode, while others like:

`# Import NI signing key for subsequent opkg operations
/usr/sbin/chroot /mnt/userfs/ opkg-key populate`

make use of chroot in the SystemImage postinst script.

Reply:
Yeah, that distinction is what I'm asking about. If you run this script while chrooted into /mnt/userfs, then I *think* the `/etc/fstab` you're modifying will be the *runmode* fstab (assuming that we don't symlink that file between safemode and runmode). So I would guess that the script, as written, might write the `/mnt/userfs` fstab entry into the runmode fstab file - which isn't a sane entry for the runmode context.

I might be mistaken about how you're calling the script though. You can quickly check that my concerns are invalid by checking the value of `/mnt/userfs/etc/fstab` after the script is run. As long as it doesn't get the `/mnt/userfs` entry, I'm satisfied.

Reply:
I've checked and confirm that the scripts only writes to /mnt/userfs/etc/fstab (and fw_env.config) and that after the script was executed, no /mnt/userfs/mnt/userfs exists.

Reply:
OK; but does `/mnt/userfs/etc/fstab` contain an entry for `/mnt/userfs`?

Reply:
I realized I got confused by myself with the other commit in review-board. This script runs as S00nisdbootconfig startup script instead of by postinst. It will not be executed by chroot. Only the /etc/fstab in safemode gets modified from this script.

Nope, /mnt/userfs/etc/fstab does not contain entry for /mnt/userfs.